lib/puppet/indirector/json.rb

```diff
@@ -52,7 +52,7 @@ def load_json_from_file(file, key)
     json = nil
 
     begin
-      json = File.read(file)
+      json = File.read(file).force_encoding(Encoding::UTF_8)
```

Comment:
I think we're saying the same thing, but here's an example. Ruby on Windows sets its default external encoding to match the current code page. So suppose we're in code page 1252 and we write out the [euro symbol](http://www.fileformat.info/info/unicode/char/20aC/index.htm). In CP1252, the euro character is a single byte 0x80. When ruby writes the character to disk, it doesn't change the byte representation, so the file on disk contains a single byte 0x80. When we read the file in, we read the contents back in as a CP1252 encoded string (since that's the default external encoding). The string at that point is still valid. If we then force the encoding to UTF-8, we essentially "corrupt" the string since the sequence 0x80 is not a valid UTF-8 encoded string. For example, given:

```ruby
# -*- coding: windows-1252 -*-

puts "encoding internal=#{Encoding.default_internal}, external=#{Encoding.default_external}"

euro = "\u20AC".encode(Encoding::CP1252)
puts "#{euro}=#{euro.unpack('H*')[0]}"

File.open('foo.json', 'w') do |f|
  f.print euro
end

str = File.read('foo.json')
puts "#{str}=#{str.unpack('H*')[0]}"

str.force_encoding('utf-8')
puts "#{str}=#{str.unpack('H*')[0]}"
```

Running this with codepage 1252:

```
C:\users\albert> ruby .\encoding.rb
encoding internal=, external=Windows-1252
€=80
€="80"
�="80"
```

Note how the last string after we forced the encoding is invalid.

The only time we can force encoding UTF-8 is if we're sure the data was written out in UTF-8. So that's why I was saying we should modify the `save` method to explicitly call `String#encode('utf-8')`. For example, if we change it to:

```
  f.print euro.encode('utf-8')
```

Then the `force_encoding` is required:

```
C:\users\albert> ruby .\encoding.rb
encoding internal=, external=Windows-1252
€=80
â‚¬=e282ac
€=e282ac
```

where e2, 82, ac is the correct UTF-8 encoding of the euro symbol.